### PR TITLE
jar: replace go-dep-parser's jar package

### PR DESCRIPTION
**Pull request by hdonnay**
These commits remove the external jar parser and replace it with an internal one.

The previous one made unconfigurable HTTP requests and seemed to hold onto memory.

### CODE REVIEW

**Comment by crozzy** [comment]
I'm getting this error when trying to run the integration tests:
go test -timeout 30s -run ^TestParse$ github.com/quay/claircore/java/jar -tags integration
```
    jar_test.go:47: found jar: apache-cassandra-4.0.0/lib/netty-tcnative-boringssl-static-2.0.36.Final.jar
    jar_test.go:51: unexpected EOF
    jar_test.go:73: unexpected EOF
FAIL
FAIL	github.com/quay/claircore/java/jar	0.444s
FAIL
```
I tried re: go generating the jar files but it didn't seem to make a difference.

**Review by crozzy — COMMENTED** [review]
Just a few smaller comments

**Comment by crozzy** [comment]
> I'm getting this error when trying to run the integration tests:
> go test -timeout 30s -run ^TestParse$ github.com/quay/claircore/java/jar -tags integration
> 
> ```
>     jar_test.go:47: found jar: apache-cassandra-4.0.0/lib/netty-tcnative-boringssl-static-2.0.36.Final.jar
>     jar_test.go:51: unexpected EOF
>     jar_test.go:73: unexpected EOF
> FAIL
> FAIL	github.com/quay/claircore/java/jar	0.444s
> FAIL
> ```
> 
> I tried re: go generating the jar files but it didn't seem to make a difference.

I guess it's passing in the CI so it must be something system specific

File: java/jar/jar_test.go [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+		t.Error(err)
+	}
+	ps, err := Parse(ctx, name, z)
+	switch {
+	case errors.Is(err, nil):
+		for _, p := range ps {
+			t.Log(p.String())
+		}
+	case errors.Is(err, ErrUnidentified):
+		t.Log(err)
```
[the remark this answers is not in the document]
Comment (hdonnay) [reply java/jar/jar_test.go:104]: Yeah, that was a leftover from writing the test.

File: java/jar/jar_test.go [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+			continue
+		}
+		z, err := zip.NewReader(bytes.NewReader(buf.Bytes()), h.Size)
+		if err != nil {
+			t.Error(err)
+			continue
+		}
+		ps, err := Parse(ctx, h.Name, z)
+		switch {
+		case errors.Is(err, nil):
```
[the remark this answers is not in the document]
Comment (hdonnay) [reply java/jar/jar_test.go:62]: I don't think so. The heuristic looks for things in a defined order, and the [WalkDir](https://pkg.go.dev/io/fs#WalkDir) documentation says it traverses in lexical order.

**Comment by hdonnay** [comment]
Addressed comments and added checksums to the fetched files to hopefully prevent weirdness around incomplete downloads.

**Review by crozzy — APPROVED** [review]
LGTM :+1:

**Pull request merged**